lib/ret/avatar.ex

```diff
@@ -70,24 +70,41 @@ defmodule Ret.Avatar do
     end)
   end
 
+  defp avatar_listing_to_collapsed_files(%{parent_avatar_listing: nil} = listing),
+    do: listing |> Map.take(Avatar.file_columns())
+
+  defp avatar_listing_to_collapsed_files(%{parent_avatar_listing: parent} = listing) do
+    parent
+    |> Repo.preload(Avatar.file_columns())
+    |> Map.take(Avatar.file_columns())
+    |> Map.merge(listing |> Map.take(Avatar.file_columns()), fn
+      _k, v1, nil -> v1
+      _k, _v1, v2 -> v2
+    end)
+  end
```

Comment:
Seems like this can be DRYed some more with `avatar_to_collapsed_files` now that they are in the same module. At least now you can just do `@file_columns` instead of `Avatar.file_columns()`, but everything past line 77 here is identical in `avatar_to_collapsed_files`. Maybe just nitpicking now though, not a big deal.

Reply:
Its the pattern matching of the argumetns that differs, could break out just the body but ends up being basically just as much code anyway. Will filx `@file_columns` though